src/bgw/job.c

```diff
@@ -974,12 +974,13 @@ ts_bgw_job_check_max_retries(BgwJob *job)
 }
 
 void
-ts_bgw_job_permission_check(BgwJob *job)
+ts_bgw_job_permission_check(BgwJob *job, const char *cmd)
 {
 	if (!has_privs_of_role(GetUserId(), job->fd.owner))
 		ereport(ERROR,
 				(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
-				 errmsg("insufficient permissions to alter job %d", job->fd.id)));
+				 errmsg("insufficient permissions to %s job %d", cmd, job->fd.id),
+				 errhint("You need to belong to the group that owns the job.")));
```

Comment:
The hint doesn't seem very clear to me. Should we mention "user group role"?

Reply:
> The hint doesn't seem very clear to me. Should we mention "user group role"?

We could, but I think we might actually print the actual group name and user as a detailed message as well to make it very clear.

Reply:
Added a more elaborate error detail message.